sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -634,4 +648,32 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
       },
     });
   }
+
+  /**
+   * Check if the given institution has already a program with the given SABC code.
+   * @param institutionId id of the institution to have the programs retrieved.
+   * @param sabcCode SABC code.
+   * @param programId programId in case it is an update. It will be ignored in case of `undefined`.
+   * @returns true in case it has already a SABC code for the institution.
+   */
+  async hasExistingProgramWithSameSABCCode(
+    institutionId: number,
+    sabcCode: string,
+    programId: number,
```

Comment:
programid can be null from the consumer right. so `programId?: number,`

Reply:
yes, right. Added type `undefined` to programId. Thanks!